src/main/scala/ai/privado/languageEngine/java/passes/config/JavaPropertyLinkerPass.scala

```diff
@@ -66,11 +67,29 @@ class JavaPropertyLinkerPass(cpg: Cpg) extends PrivadoParallelCpgPass[JavaProper
         builder.addEdge(propertyNode, value, EdgeTypes.IS_USED_AT)
         builder.addEdge(value, propertyNode, EdgeTypes.ORIGINAL_PROPERTY)
       }
+
+    val annotatedMethodsList = annotatedMethods()
+
+    annotatedMethodsList
+      .filter { case (key, _) => propertyNode.name == key.code.slice(3, key.code.length - 2) }
+      .foreach { case (_, value) =>
+        builder.addEdge(propertyNode, value, EdgeTypes.IS_USED_AT)
+        builder.addEdge(value, propertyNode, EdgeTypes.ORIGINAL_PROPERTY)
+      }
   }
 
+  /** List of all methods annotated with Spring's `Value` annotation
+    */
+  private def annotatedMethods(): List[(AnnotationParameterAssign, Method)] = cpg.annotation
+    .fullName(".*Value.*")
+    .filter(_.method.nonEmpty)
+    .filter(_.parameterAssign.nonEmpty)
+    .map { x => (x.parameterAssign.next(), x.method.next()) }
+    .toList
+
   /** List of all parameters annotated with Spring's `Value` annotation, along with the property name.
     */
-  def annotatedParameters(): List[(MethodParameterIn, String)] = cpg.annotation
+  private def annotatedParameters(): List[(MethodParameterIn, String)] = cpg.annotation
     .fullName("org.springframework.*Value")
```

Comment:
The full name may not always contain the `springframework` info. Check on Annotation name